util/ubuntu-dependencies

```diff
@@ -0,0 +1,90 @@
+#!/bin/bash
+# This script installs streisand's dependencies on an Ubuntu 16.04
+# system; Debian jessie has been lightly tested. It installs system
+# packages for binary dependencies, and installs the rest in the
+# python directories in $HOME/.local.
+
+# We default to a regular, interactive upgrade:
+quiet=
+
+# It's safe to run this script multiple times.
+#
+# Please don't paste this directly into an interactive shell; if you
+# do, I will instantly close any bugs you file.
+#
+# If you're brave: this script can proceed mostly without interaction if:
+#
+# 1) "sudo echo" will succeed (i.e., you have root without a prompt now)
+#
+# 2) uncomment the following line:
+
+# quiet='--yes'
+
+# If you *really* want no interaction (for example, overwriting local
+# config files without prompting), then uncomment this:
+# export DEBIAN_FRONTEND=noninteractive
+
+### No options below this line. ###
+
+set -e
+set -x
+
+function our_pip_install () {
+    pip install --user --upgrade "$@"
+}
+
+# Hopefully, the blank lines following these statements will answer "yes" if
+# somebody's pasting.
+
+sudo apt-get update
+
+sudo apt-get $quiet upgrade
+
+
+
+# We want word splitting.
+# shellcheck disable=SC2046
+sudo apt-get $quiet install $(cat <<EOF
+build-essential
+python-dev
+python-setuptools
+python-pip
+python-cffi libffi-dev
+EOF
+)
+
+
+# Debian doesn't have python-nacl.
```

Comment:
It seems like there are a couple Debian related divergences. I think it would be cleaner to try and detect if the system is Debian (maybe with `lsb_release`?) and if so then do the Debian required steps right-off instead of this approach of trying the Ubuntu way and falling back if it fails assuming its a Debian system.